civicase.php

```diff
@@ -443,8 +443,20 @@ function civicase_civicrm_postProcess($formName, &$form) {
   if (in_array($formName, $specialForms)) {
     $urlParams = parse_url(htmlspecialchars_decode($form->controller->_entryURL), PHP_URL_QUERY);
     parse_str($urlParams, $urlParams);
-    if (!empty($urlParams['draft_id'])) {
-      civicrm_api3('Activity', 'delete', ['id' => $urlParams['draft_id']]);
+
+    $ifDownloadDocumentButtonClicked = array_key_exists('_qf_PDF_upload', $form->getVar('_submitValues')['buttons']);
+
+    if ($ifDownloadDocumentButtonClicked && !empty($urlParams['draft_id'])) {
+      $completedActivityStatusId = civicrm_api3('OptionValue', 'get', [
+        'sequential' => 1,
+        'option_group_id' => "activity_status",
+        'name' => "Completed",
+      ])['values'][0]['value'];
+
+      civicrm_api3('Activity', 'create', [
+        'id' => $urlParams['draft_id'],
+        'status_id' => $completedActivityStatusId,
```

Comment:
We can pass `Completed` directly to the `status_id` parameter, so we can remove the code for fetching the id.